lib/autosign/validator.rb

```diff
@@ -1,88 +1,26 @@
-    # You must set the NAME constant in the sublcass
-    def name
-      self.class::NAME
-    end
-
-    # define how a validator actually validates the request.
-    # This must be implemented by validators which inherit from the
-    # Autosign::Validator class.
-    #
-    # @param challenge_password [String] the challenge_password OID from the certificate signing request. The challenge_password field is the same setting as the "challengePassword" field in a `csr_attributes.yaml` file when the CSR is generated. In a request using a JSON web token, this would be the serialized token.
-    # @param certname [String] the common name being requested in the certificate signing request. Treat the certname as untrusted. This is user-submitted data that you must validate.
-    # @param raw_csr [String] the encoded X509 certificate signing request, as received by the autosign policy executable. This is provided as an optional extension point, but your validator may not need to use it.
-    # @return [True, False] return true if the certificate should be signed, and false if you cannot validate the request successfully.
-    def perform_validation(_challenge_password, _certname, _raw_csr)
-      # override this after inheriting
-      # should return true to indicate success validating
-      # or false to indicate that the validator was unable to validate
-      raise NotImplementedError
-    end
-
-    # wrapper method that wraps input validation and logging around the perform_validation method.
-    # Do not override or use this class in child classes. This is the class that gets called
-    # on validator objects.
-    def validate(challenge_password, certname, raw_csr)
-      @log.debug "attempting to validate using #{name}"
-      raise unless challenge_password.is_a?(String)
-      raise unless certname.is_a?(String)
-
-      case perform_validation(challenge_password, certname, raw_csr)
-      when true
-        @log.debug 'validated successfully'
-        @log.info  "Validated '#{certname}' using '#{name}' validator"
-        true
-      when false
-        @log.debug 'validation failed'
-        @log.debug "Unable to validate '#{certname}' using '#{name}' validator"
-        false
-      else
-        @log.error 'perform_validation returned a non-boolean result'
-        raise 'perform_validation returned a non-boolean result'
-      end
-    end
-
+  module Validator
+    
     # @return [Array] - A list of all the validator classes
     # @param list [Array] - a list of validators to use, uses the settings list by default
     # This returns a list of validators that were specified by the user and the exact
     # order they want the validation to procede.
-    def self.validators(list = nil)
-      validation_order = list || Autosign::Config.new.settings['general']['validation_order']
+    def self.validation_order(settings = Autosign::Config.new.settings, list = nil)
```

Comment:
Why the method name change? It makes it a little harder to read what's going on later.

Reply:
Validator.validators looked weird and this more closely represents what the code is actually doing.  It returns a list of ordered validator classes.